src/Symfony/Component/Serializer/Encoder/ChainDecoder.php

```diff
@@ -67,9 +67,13 @@ private function getDecoder(string $format, array $context): DecoderInterface
             return $this->decoders[$this->decoderByFormat[$format]];
         }
 
+        $cache = true;
         foreach ($this->decoders as $i => $decoder) {
+            $cache = $cache && !$decoder instanceof ContextAwareDecoderInterface;
```

Comment:
Sure you want to override the value defined in the outer scope? Then it would result in `false` for all decoders after it has been `false` for one.

Reply:
Sure: as soon as a non-context-aware decoder is found, we cannot cache any of them for the specified $format.